js/filter.js

```diff
@@ -1,70 +1,76 @@
+  } else {
+    return parseInt(newArray.offer[filterValue], 10) === parseInt(window.activate.formFiltersNode[objectValue].value, 10);
+  }
+};
 
-      let simmillarPinsArray = array;
+window.filter = {
+  updateSimillarPins: (array) => {
 
-      simmillarPinsArray = simmillarPinsArray.filter((pinSimmillar) => {
-        if (window.activate.formFiltersNode[`housing-type`].value === `any`) {
-          return simmillarPinsArray;
-        } else {
-          return pinSimmillar.offer.type === window.activate.formFiltersNode[`housing-type`].value;
-        }
-      });
+    let simmillarPinsArray = array;
 
-      simmillarPinsArray = simmillarPinsArray.filter((pinSimmillar) => {
-        return containsValue(`housing-rooms`, `rooms`, simmillarPinsArray, pinSimmillar);
-      });
+    const filterPinsByType = (pinSimmillar) => {
+      if (window.activate.formFiltersNode[`housing-type`].value === FILTER_DEFAULT_VALUE) {
+        return simmillarPinsArray;
+      } else {
+        return pinSimmillar.offer.type === window.activate.formFiltersNode[`housing-type`].value;
+      }
+    };
 
-      simmillarPinsArray = simmillarPinsArray.filter((pinSimmillar) => {
-        return containsValue(`housing-guests`, `guests`, simmillarPinsArray, pinSimmillar);
-      });
+    const filterPinsByRooms = (pinSimmillar) => {
+      return containsValue(`housing-rooms`, `rooms`, simmillarPinsArray, pinSimmillar);
+    };
 
-      simmillarPinsArray = simmillarPinsArray.filter((pinSimmillar) => {
-        if (window.activate.formFiltersNode[`housing-price`].value === `low`) {
+    const filterPinsByGuests = (pinSimmillar) => {
+      return containsValue(`housing-guests`, `guests`, simmillarPinsArray, pinSimmillar);
+    };
+
+    const filterPinsByPrice = (pinSimmillar) => {
+      switch (window.activate.formFiltersNode[`housing-price`].value) {
+        case `low`:
           return pinSimmillar.offer.price < RoomPrice.low;
-        } else if (window.activate.formFiltersNode[`housing-price`].value === `middle`) {
+        case `middle`:
           return pinSimmillar.offer.price >= RoomPrice.low && pinSimmillar.offer.price <= RoomPrice.high;
-        } else if (window.activate.formFiltersNode[`housing-price`].value === `high`) {
+        case `high`:
           return pinSimmillar.offer.price > RoomPrice.high;
-        } else {
+        default:
           return simmillarPinsArray;
-        }
-      });
+      }
+    };
 
-      checkBoxes.forEach((element) => {
-        if (element.checked) {
-          simmillarPinsArray = simmillarPinsArray.filter((pinSimmillar) => {
-            return pinSimmillar.offer.features.includes(element.value);
-          });
-        }
+    const filterPinsByFeatures = function (pinSimmillar) {
+      return !checkBoxes.some(function (element) {
+        return element.checked && !pinSimmillar.offer.features.includes(element.value);
       });
+    };
+
+    const newArray = simmillarPinsArray.filter(filterPinsByType)
```

Comment:
Кажется так описательнее
```suggestion
    const filteredOffersArray = simmillarPinsArray.filter(filterPinsByType)
```